lldb/bindings/interface/SBTargetExtensions.i

```diff
@@ -79,11 +79,11 @@ STRING_EXTENSION_LEVEL_OUTSIDE(SBTarget, lldb::eDescriptionLevelBrief)
                         module = self.sbtarget.GetModuleAtIndex(idx)
                         if module.uuid == key:
                             return module
-                elif type(key) is re.SRE_Pattern:
+                elif isinstance(key, type(re.compile(''))):
                     matching_modules = []
                     for idx in range(num_modules):
                         module = self.sbtarget.GetModuleAtIndex(idx)
-                        re_match = key.search(module.path.fullpath)
+                        re_match = key.search(module.file.fullpath)
```

Comment:
Since this was broken, and obviously not used, I'd like to also change this from `fullpath` to `basename`. Opinions?

Reply:
Does this mean you'd be able to search for modules just by the module's base name instead of the complete path? e.g. You could find libc++ with `libc++` instead of `/usr/lib/libc++.dylib` or whatever.

If so, I'd be supportive of that kind of change.

Reply:
note that lookup supports string keys, one can write `target.module["libc++.dylib"]`

Reply:
I'll leave such a change for a follow up.